ql/src/java/org/apache/hadoop/hive/ql/metadata/MaterializedViewMetadata.java

```diff
@@ -32,16 +36,19 @@
 import static java.util.Collections.unmodifiableSet;
 
 public class MaterializedViewMetadata {
+  private static final Logger LOG = LoggerFactory.getLogger(MaterializedViewMetadata.class);
+
   final CreationMetadata creationMetadata;
 
   MaterializedViewMetadata(CreationMetadata creationMetadata) {
     this.creationMetadata = creationMetadata;
   }
 
   public MaterializedViewMetadata(
-          String catalogName, String dbName, String mvName, Set<SourceTable> sourceTables, String validTxnList) {
+          String catalogName, String dbName, String mvName, Set<SourceTable> sourceTables,
+          MaterializationSnapshot snapshot) {
     this.creationMetadata = new CreationMetadata(catalogName, dbName, mvName, toFullTableNames(sourceTables));
-    this.creationMetadata.setValidTxnList(validTxnList);
+    this.creationMetadata.setValidTxnList(snapshot.asJsonString());
```

Comment:
I don't recommend changing it because of backward compatibility reasons. My original idea was to introduce a new field to store the snapshotIds however that requires hms backend db schema change and thrift api change which I would like to avoid.